### PR TITLE
Update Long Haul and Image Build Pipeline to use the new Direct Method Sender Module

### PR DESCRIPTION
**Pull request by yophilav**
- Remove the old Direct Method Cloud Sender module
- Modify Long Haul pipeline to handle a new deployment for the new Direct Method Sender module
- Set the default value of the Direct Method Sender to be compatible with the old E2E pipeline
- Fix a broken reporting endpoint for Long Haul pipeline

### CODE REVIEW

**Comment by philipktlin** [comment]
Please add description to your PR.

File: edge-modules/DirectMethodSender/DirectMethodCloudSender.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public class DirectMethodCloudSender : DirectMethodSenderClient
+    {
+        readonly ServiceClient serviceClient;
+
+        private DirectMethodCloudSender(
+            ServiceClient serviceClient,
+            ILogger logger)
+            : base(logger)
+        {
+            this.serviceClient = serviceClient;
```
Comment (philipktlin) [inline edge-modules/DirectMethodSender/DirectMethodCloudSender.cs]:
```suggestion
            this.serviceClient = Preconditions.CheckNotNull(serviceClient);
```

File: e2e_deployment_files/long_haul_deployment.template.json [hunk 2]
```diff
@@ -370,8 +376,8 @@
           "TempSensorToTempFilter": "FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint(\"/modules/tempFilter/inputs/input1\")",
           "TempFilterToCloud2": "FROM /messages/modules/tempFilter2/outputs/* INTO $upstream",
           "TempSensorToTempFilter2": "FROM /messages/modules/tempSensor2/outputs/temperatureOutput INTO BrokeredEndpoint(\"/modules/tempFilter2/inputs/input1\")",
-          "upstreamDirectMethodCloudSender1": "FROM /messages/modules/DirectMethodCloudSender/outputs/output1 INTO $upstream",
-          "upstreamDirectMethodCloudSender2": "FROM /messages/modules/DirectMethodCloudSender2/outputs/output1 INTO $upstream"
+          "upstreamDirectMethodSender1": "FROM /messages/modules/DirectMethodSender/outputs/output1 INTO $upstream",
```
Comment (philipktlin) [inline e2e_deployment_files/long_haul_deployment.template.json]: nit: I don't know why previously name it like this, but should we name it as "DirectMethodSender1ToUpstream"?

**Comment by yophilav** [comment]
The PR was built into artifact (buildId=27506315, 20191219.4) and tested on Long Haul (#20191219.2) and E2E (Release-1856). Both of the test passed.

**Pull request merged**